### ISSUE TITLE
Fatal error when trying to move completed torrent

### ISSUE THREAD

**Issue by tomaszni** [elgatito/elementum#13]
Elementum plugin 0.1.22, running on KODI 17.6 @ OSMC 2018.12-1 on RaspberryPI 3B+

Storage set up as file, with option to move completed torrents to specified destination. On torrent completion, following error occurs (leading to Elementum restart, files not getting moved, and torrent info missing):

```
08:45:58.303 T:1371521792  NOTICE: [plugin.video.elementum] INFO  bittorrent   ▶ func1            Deleting fast resume data at /home/osmc/.kodi/temp/elementum_torrents/fad678bb2be51b2aa6661ff844d126a04c380f93.fastresume
08:45:58.304 T:1371521792  NOTICE: [plugin.video.elementum] INFO  bittorrent   ▶ func1            Deleting torrent file at  /home/osmc/.kodi/temp/elementum_torrents/fad678bb2be51b2aa6661ff844d126a04c380f93.torrent
08:45:58.305 T:1371521792  NOTICE: [plugin.video.elementum] panic: interface conversion: interface {} is string, not int
08:45:58.306 T:1371521792  NOTICE: [plugin.video.elementum]
08:45:58.307 T:1371521792  NOTICE: [plugin.video.elementum] goroutine 86 [running]:
08:45:58.308 T:1371521792  NOTICE: [plugin.video.elementum] github.com/ElementumOrg/libtorrent-go.SwigcptrFileStorage.FilePath(0x61cc0c08, 0x30decb8, 0x1, 0x1, 0x61cc0c01, 0x30decb8)
08:45:58.309 T:1371521792  NOTICE: [plugin.video.elementum]     _libtorrent_swig.go:20687 +0x124
08:45:58.310 T:1371521792  NOTICE: [plugin.video.elementum] github.com/elgatito/elementum/bittorrent.(*Service).downloadProgress.func1(0x34878f0, 0x28, 0x336adf4, 0x3780a00, 0x39, 0x336ae10, 0x329a180, 0x129a570, 0x34e2020, 0x0, ...)
08:45:58.311 T:1371521792  NOTICE: [plugin.video.elementum]     /go/src/github.com/elgatito/elementum/bittorrent/service.go:1195 +0x784
08:45:58.312 T:1371521792  NOTICE: [plugin.video.elementum] github.com/elgatito/elementum/bittorrent.(*Service).downloadProgress(0x329a180)
08:45:58.313 T:1371521792  NOTICE: [plugin.video.elementum]     /go/src/github.com/elgatito/elementum/bittorrent/service.go:1268 +0x718
08:45:58.314 T:1371521792  NOTICE: [plugin.video.elementum] created by github.com/elgatito/elementum/bittorrent.NewService
08:45:58.315 T:1371521792  NOTICE: [plugin.video.elementum]     /go/src/github.com/elgatito/elementum/bittorrent/service.go:113 +0x280
08:45:58.355 T:1371521792  NOTICE: [plugin.video.elementum] System information: linux_armv7 4.14.78-4-osmc
08:45:58.356 T:1371521792  NOTICE: [plugin.video.elementum] Kodi build version: 17.6
08:45:58.356 T:1371521792  NOTICE: [plugin.video.elementum] OS type: Linux
08:45:58.357 T:1371521792  NOTICE: [plugin.video.elementum] uname: ('Linux', 'osmc', '4.14.78-4-osmc', '#1 SMP PREEMPT Wed Dec 12 17:58:11 UTC 2018', 'armv7l', '')
08:46:03.365 T:1371521792  NOTICE: [plugin.video.elementum] elementumd: starting elementumd
```

Seems that https://github.com/elgatito/elementum/blob/v0.1.22/bittorrent/service.go#L1195 is the fault.
Maybe should be `filePath := torrentInfo.Files(i).FilePath()` instead? I don't know Go, just my intuition ;)

**Comment by elgatito** [comment]
Thanks for this catch.
Will be fixed in the next version.

**Comment by tomaszni** [comment]
I think you missed the fix in the latest version

**Comment by elgatito** [comment]
@tomaszni it should be fixed in 0.1.23

**Comment by tomaszni** [comment]
Yeah, my bad - I just looked at the commit descriptions, and missed the bundled fixes :)

Closing as fixed
